aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -130,6 +132,52 @@ The following examples use the <xref:Microsoft.AspNetCore.Server.Kestrel.Core> n
 using Microsoft.AspNetCore.Server.Kestrel.Core;
 ```
 
+Kestrel options, which are configured in C# code in the following examples, can also be set using a [configuration provider](xref:fundamentals/configuration/index). For example, the File Configuration Provider can load Kestrel configuration from a file (*appsettings.json* or *appsettings.{Environment}.json*):
+
+```json
+{
+  "Kestrel": {
+    "Limits": {
+      "MaxConcurrentConnections": 100,
+      "MaxConcurrentUpgradedConnections": 100
+    },
+    "DisableStringReuse": true
+  }
+}
+```
+
+Use **one** of the following approaches:
+
+* Inject an instance of `IConfiguration` into the `Startup` class from the `Microsoft.Extensions.Configuration` namespace (`Configuration` in the following example). In `Startup.ConfigureServices`, load the `Kestrel` section of configuration into Kestrel's configuration:
```

Comment:
parenthetical clauses don't MT, so try not to use them.

Reply:
They’re all over the repo ... hundreds of uses ... this is the first time I’ve heard that.

Reply:
Now you know. I'm working on getting rid of them. Sometimes they can't be avoided, but when they can, don't use them (unless they can't be avoided (or it's a special case)).

Reply:
>  Inject an instance of `IConfiguration` into the `Startup` class from the `Microsoft.Extensions.Configuration` namespace (`Configuration` in the following example). 

That's really hard to read and is steve speak. Can't you let the code do the talking? TMI here

 Inject an instance of `IConfiguration` into the `Startup` class.  The following example ....

add notes about namespace if that's useful.

Reply:
// requires using `Microsoft.Extensions.Configuration`

Reply:
Let's get rid of the steve english version of the code and replace it with

Inject an instance of `IConfiguration` into the `Startup` class .

The provide a snippet showing that. Feel free to add to the code
// requires using ....

Reply:
Make 2 bullets here and show all the code.

* Inject an instance of `IConfiguration` into the `Startup` class.
* . In `Startup.ConfigureServices`, load the `Kestrel` section of configuration into Kestrel's configuration:

Reply:
* Doesn't work as two bullets because the other bullet is an either-or scenario. I can use a list there there tho.
* I'll move the namespace into the comment `using`, but the "requires" isn't necessary.
* I'll fix the indentation, too, of the code examples on the next commit.